### PR TITLE
Styled TextFieldIconButton

### PR DESCRIPTION
**Pull request by guilean**
**Checklist**

- [x] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://trello.com/c/95Nr1LgN/94-migrate-composer-forms-to-styled-components

### CODE REVIEW

File: packages/styled-components/src/components/primitives/Form/TextFieldIcon/TextFieldIcon.tsx [hunk 1]
```diff
@@ -43,7 +45,7 @@ export const TextFieldIcon = ({
   }, []);
 
   return (
-    <StyledWrapper>
+    <StyledWrapper css={wrapperCustomCss}>
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Form/TextFieldIcon/TextFieldIcon.tsx:48]: no className?

File: packages/styled-components/src/components/primitives/Form/TextFieldIconButton/TextFieldIconButton.types.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  required?: boolean;
+  disabled?: boolean;
+  invalid?: boolean;
+  type?: string;
+  value?: string | number;
+  iconPosition?: IconPositions;
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
+  onClick?: () => void;
+  buttonContent?: React.ReactNode;
+  icon?: React.ReactNode;
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Form/TextFieldIconButton/TextFieldIconButton.types.ts:21]: use `React.ChangeEvent` and remove the import or import `ReactNode`

**Review by tkanzakic — CHANGES_REQUESTED** [review]

File: packages/styled-components/src/components/primitives/Form/TextFieldIcon/TextFieldIcon.tsx (hunk 1 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/Form/TextFieldIcon/TextFieldIcon.tsx]: Added type and test

File: packages/styled-components/src/components/primitives/Form/TextFieldIconButton/TextFieldIconButton.types.ts (hunk 2 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/Form/TextFieldIconButton/TextFieldIconButton.types.ts]: Done

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**